### PR TITLE
SCU rebalance

### PR DESCRIPTION
**Pull request by TurinturambarFAF**
Nerfs the raw SCU HP and dps to the level of a vanilla ACU and also nerfs the raw SCU bp from 56 to 40 and compensates these changes by buffs of the respective upgrades.

See: #4642

### CODE REVIEW

**Comment by Garanas** [comment]
What are the ink files 🤔 ?

**Comment by Tagada14** [comment]
Please fix some of the formatting issues and the ink files. Also, shouldn't the Cybran have 10k hp if we are going with the same hp as the ACU theme?

**Comment by Tagada14** [comment]
Please resolve the conflict so that I can merge the PR

**Comment by ComradeStryker** [comment]
Won't this affect how much HP these units gain when they vet?

These changes will affect the 2 factions with shieldable SACUs more than the other two.
In the long run, they're losing more HP, as for example, the Seraphim's SACU's shield doesn't vet with the unit.

---

~ Stryker

**Comment by Tagada14** [comment]
It will have slight effects on the units but it really doesn't matter. The difference will be negligible.

File: units/UAL0301/UAL0301_unit.bp [hunk 1]
```diff
@@ -494,7 +494,7 @@ UnitBlueprint {
             Icon = 'ss',
             Name = '<LOC enhancements_0031>Reacton Refractor',
             NewDamageRadiusMod = 3.5,
-
+            NewDamageMod = 200,
```
[the remark this answers is not in the document]
Comment (Tagada14) [reply units/UAL0301/UAL0301_unit.bp:497]: Actually you could change this:
`wep:AddDamageMod(-self:GetBlueprint().Enhancements['RightReactonCannon'].NewDamageMod)`
to:
`wep:AddDamageMod(-self:GetBlueprint().Enhancements['StabilitySuppressant'].NewDamageMod)`

File: units/UEL0301/UEL0301_script.lua [hunk 2]
```diff
@@ -161,9 +161,11 @@ UEL0301 = Class(CommandUnit) {
         elseif enh =='AdvancedCoolingUpgrade' then
             local wep = self:GetWeaponByLabel('RightHeavyPlasmaCannon')
             wep:ChangeRateOfFire(bp.NewRateOfFire)
+            wep:AddDamageMod(bp.UpgradeDamageMod)
         elseif enh =='AdvancedCoolingUpgradeRemove' then
             local wep = self:GetWeaponByLabel('RightHeavyPlasmaCannon')
             wep:ChangeRateOfFire(self:GetBlueprint().Weapon[1].RateOfFire or 1)
+            wep:AddDamageMod(-bp.UpgradeDamageMod)
```
Comment (Tagada14) [inline units/UEL0301/UEL0301_script.lua:168]: ` wep:AddDamageMod(-self:GetBlueprint().Enhancements['AdvancedCoolingUpgrade'].UpgradeDamageMod)`

File: units/XSL0301/XSL0301_unit.bp [hunk 3]
```diff
@@ -399,6 +399,7 @@ UnitBlueprint {
                 'Back_Upgrade',
             },
             Slot = 'Back',
+            UpgradeDamageMod = 300,
```
Comment (Tagada14) [inline units/XSL0301/XSL0301_unit.bp:402]: This should be NewDamageMod

File: units/XSL0301/XSL0301_script.lua [hunk 4]
```diff
@@ -156,6 +157,7 @@ XSL0301 = Class(CommandUnit) {
             self:SetIntelRadius('Vision', bpIntel.VisionRadius or 26)
             self:SetIntelRadius('Omni', bpIntel.OmniRadius or 16)
             local wep = self:GetWeaponByLabel('LightChronatronCannon')
+            wep:AddDamageMod(-bp.NewDamageMod or 0)
```
Comment (Tagada14) [inline units/XSL0301/XSL0301_script.lua:160]: `wep:AddDamageMod(-wep:AddDamageMod(-self:GetBlueprint().Enhancements['EnhancedSensors'].NewDamageMod))`

**Review by Tagada14 — CHANGES_REQUESTED** [review]

**Comment by Tagada14** [comment]
@Garanas 
Is this fine or bad practice?
`wep:AddDamageMod(-wep:AddDamageMod(-self:GetBlueprint().Enhancements['EnhancedSensors'].NewDamageMod))`
I would like to avoid having to add values like HP, Damage etc. in both the enchancement and it's removal but I would like to know if this method if fine or not. Also perhaps there is a better way?

**Comment by Tagada14** [comment]
@TurinturambarFAF Can you allow me to push to your Repo so that I can update this branch with the changes fixing all those things I mentioned in the review?
Also, next time please at least test the changes, half of the damage upgrades were broken XD

**Comment by Garanas** [comment]
> @Garanas Is this fine or bad practice? `wep:AddDamageMod(-wep:AddDamageMod(-self:GetBlueprint().Enhancements['EnhancedSensors'].NewDamageMod))` I would like to avoid having to add values like HP, Damage, etc. in both the enhancement and its removal but I would like to know if this method is fine or not. Also perhaps there is a better way?


I'm not too familiar with how enhancements work, but it seems to me it is best to do the same thing that the gun upgrade of ACUs do in terms of coding style.

**Pull request closed without merging**